### PR TITLE
adding go modules

### PR DESCRIPTION
**Pull request by sciabarracom**
This PR add go modules to be able to use CounterGo as a package (I am using it in a serverless app)

### CODE REVIEW

**Comment by ChizhovVadim** [comment]
I guess single go.mod file is enough for all packages. Whatever all this packages will have same version. I am going to add it.

**Comment by sciabarracom** [comment]
Actually not. Each package needs a go.mod that identifies the package.
I am trying to use CounterGo as a library for a serverless app and I faced this problem.

**Comment by ChizhovVadim** [comment]
I did not find such an approach in public golang repositories.
Another possible way is to use own github repository for each package.


пт, 11 сент. 2020 г. в 10:14, Michele Sciabarra <notifications@github.com>:

**Comment by sciabarracom** [comment]
I actually already forked and built your code here: https://github.com/sciabarracom/CounterGo but I had to add all the go.mod and a lot of replacements to make it work. 

This PR allows to use your engine as as a library. I am using it as an engine for a serverless action and I do not want to fork it but usa as a library. 

If you try to build this:

https://github.com/openwhisk-blog/whisk-chess/

you see the problems and why I sent this PR

```
 go build .
go: finding module for package github.com/ChizhovVadim/CounterGo/eval
go: finding module for package github.com/ChizhovVadim/CounterGo/uci
main.go:8:2: module github.com/ChizhovVadim/CounterGo@latest found (v2.1.0+incompatible), but does not contain package github.com/ChizhovVadim/CounterGo/eval
main.go:9:2: module github.com/ChizhovVadim/CounterGo@latest found (v2.1.0+incompatible), but does not contain package github.com/ChizhovVadim/CounterGo/uci
```

Basically to use as a library I need to put the go.mod because otherwise go will not recognize as a module.
Furthermore, because many methods in UCI are private, I added a Play method in order to be able to play one move starting from any FEN position.
